codegen/src/model/SemanticVersion.test.scala

```diff
@@ -0,0 +1,32 @@
+package besom.model
+
+//noinspection ScalaFileName
+class SemanticVersionTest extends munit.FunSuite:
+
+  Map(
+    "1.0.0" -> SemanticVersion(1, 0, 0),
+    "v1.0.0" -> SemanticVersion(1, 0, 0),
+    "0.0.1-SNAPSHOT" -> SemanticVersion(0, 0, 1, "SNAPSHOT"),
+    "v0.0.1-SNAPSHOT" -> SemanticVersion(0, 0, 1, "SNAPSHOT")
+  ).foreachEntry((input, expected) =>
+    test(s"parse $input") {
+      assertEquals(SemanticVersion.parseTolerant(input), Right(expected))
+    }
+  )
+
+  Map(
+    SemanticVersion(0, 0, 1, "SNAPSHOT") -> "0.0.1-SNAPSHOT",
+  ).foreachEntry((input, expected) =>
+    test(s"format $input") {
+      assertEquals(input.toString, expected)
+    }
+  )
+  Map(
+    SemanticVersion(0, 1, 1, "SNAPSHOT") -> "0.1.1-SNAPSHOT",
+    SemanticVersion(1, 1, 0, "SNAPSHOT") -> "1.1-SNAPSHOT",
+    SemanticVersion(1, 0, 0, "SNAPSHOT", "deadbeef") -> "1-SNAPSHOT+deadbeef",
+  ).foreachEntry((input, expected) =>
+    test(s"short format $input") {
+      assertEquals(input.toShortString, expected)
+    }
+  )
```

Comment:
lacks test of the comparing logic, doesn't it?

Reply:
done